proxy_wasm_enums.h

```diff
@@ -23,23 +23,77 @@
 
 #include <cstdint>
 
+// Severity levels for logging operations.
 enum class LogLevel : int32_t { trace, debug, info, warn, error, critical, Max = critical };
-enum class FilterStatus : int32_t { Continue = 0, StopIteration = 1 };
+
+// Enum indicating whether to continue processing of a TCP(-like) stream
+// following a callback.
+enum class FilterStatus : int32_t {
+  // The host should continue to process the stream.
+  Continue = 0,
+  // The host should suspend further processing of the stream until plugin code
+  // unpauses the stream via a call to `continueDownstream` or
+  // `continueUpstream`.
+  StopIteration = 1
+};
+
+// Enum indicating whether to continue processing of a stream following an HTTP
+// header-related callback.
 enum class FilterHeadersStatus : int32_t {
+  // The host should continue to process the stream.
   Continue = 0,
+  // The host should suspend further processing of headers until plugin code
+  // unpauses the stream via a call to `continueRequest` or `continueResponse`.
   StopIteration = 1,
+  // The host should terminate the stream.
   ContinueAndEndStream = 2,
+  // The host should suspend further processing of the stream until plugin code
+  // unpauses the stream via a call to `continueRequest` or `continueResponse`,
+  // in the meantime buffering all body bytes received.
```

Comment:
Kind of a critical bit of documentation from Envoy states:
```
  // If buffering the request causes buffered data to exceed the configured buffer limit, a 413 will
  // be sent to the user. On the response path exceeding buffer limits will result in a 500.
```

As a result, proxy_wasm_cpp_host is particularly opinionated about this state for headers -- should we mention that it's typically/often disallowed?
https://github.com/proxy-wasm/proxy-wasm-cpp-host/blob/611d5d94bb10aa5c30174e700d9031a33ba09d55/src/context.cc#L496-L501

See also discussion in https://github.com/proxy-wasm/proxy-wasm-cpp-host/issues/367

P.S. Hmmm FilterDataStatus (BODY processing) doesn't seem to have the same restriction...

Reply:
If I recall correctly, the issue we had was around pausing headers, while processing body, so this restriction was added only for headers and not to other return values.

Reply:
Added text to `StopIteration` noting that some hosts may handle it the same as `StopAllIterationAndWatermark`. Also added text mentioning that any buffering is subject to host limits.